### PR TITLE
tools: xilinx: util.tcl: Add remote flash

### PR DESCRIPTION
**Pull request by Skallwar**
## Pull Request Description

Enables remote flash capability for xilinx platform when both
`XSCT_REMOTE_HOST` and `XSCT_REMOTE_PORT` to a host running `hw_server`

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: tools/scripts/platform/xilinx/util.tcl [hunk 1]
```diff
@@ -242,7 +242,11 @@ proc upload {} {
 	set cpu [_get_processor]
 
 	# Connect to the fpga
-	connect
+  if { [info exists ::env(XSCT_REMOTE_HOST)] || [info exists ::env(XSCT_REMOTE_PORT)] } {
```
Comment (buha) [inline tools/scripts/platform/xilinx/util.tcl:245]: would && make more sense here ? they both need to be provided for it to work i assume

File: tools/scripts/platform/xilinx/util.tcl [hunk 2]
```diff
@@ -242,7 +242,11 @@ proc upload {} {
...
+    connect -host "$::env(XSCT_REMOTE_HOST)" -port "$::env(XSCT_REMOTE_PORT)"
```
Comment (buha) [inline tools/scripts/platform/xilinx/util.tcl:246]: this seems like a cool feature but we will only be known to you after merging, unless documented somewhere, like the no-OS build guide in the Xilinx sections

Comment (buha) [reply tools/scripts/platform/xilinx/util.tcl:246]: what would the flow be like? something like this?
```
export XSCT_REMOTE_HOST=11.22.33.44
export XSCT_REMOTE_PORT=5566
make run
```

Comment (Skallwar) [reply tools/scripts/platform/xilinx/util.tcl:246]: > what would the flow be like? something like this?
> 
> ```
> export XSCT_REMOTE_HOST=11.22.33.44
> export XSCT_REMOTE_PORT=5566
> make run
> ```

Yes exactly

File: tools/scripts/platform/xilinx/util.tcl (hunk 1 above)
Comment (Skallwar) [reply tools/scripts/platform/xilinx/util.tcl:245]: Yes, you are correct, I will fix this

**Comment by Skallwar** [comment]
I think this can be merged now since I've address the only comment. I will add documentation on ADI wiki as soon as this is merged if that's ok for everyone

**Comment by rbolboac** [comment]
Please perform a rebase, meanwhile we'll have a final look.

**Review by rbolboac — APPROVED** [review]

**Comment by Skallwar** [comment]
FYI a draft documenting the feature has been sent here: https://wiki.analog.com/resources/no-os/build?force_rev=1

**Pull request merged**